### DISCUSSION TITLE
One-click Installation Script for IsaacLab and IsaacSim

### DISCUSSION THREAD

**Discussion by fan-ziqi** [isaac-sim/IsaacLab#1470] [Show and tell]
# One-click Installation Script for IsaacLab and IsaacSim

Hi everyone,

I've written a one-click installation script that supports multiple IsaacLab versions with both **uv** and **conda** package managers.

## Quick Start

Install from my server

```bash
wget -O install_isaaclab.sh https://docs.robotsfan.com/install_isaaclab.sh && bash install_isaaclab.sh
```

or from [gist](https://gist.github.com/fan-ziqi/1388ffb9c853de731d970285e355d02c)

```bash
wget -O install_isaaclab.sh https://gist.githubusercontent.com/fan-ziqi/1388ffb9c853de731d970285e355d02c/raw/894aec0e0abb9f88e7a6311193fee52004d25baa/install_isaaclab.sh && bash install_isaaclab.sh
```

## Supported Versions

| Version | IsaacSim | Python | PyTorch | CUDA |
|---------|----------|--------|---------|------|
| v1.4.1 | 4.2.0.2 | 3.10 | 2.4.0 | cu118 |
| v2.3.2 | 5.1.0 | 3.11 | 2.7.0 | cu128 |
| main | 5.1.0 | 3.11 | 2.7.0 | cu128 |
| develop | 6.0.0 | 3.12 | 2.10.0 | cu128 |

## Features

- Choose between **uv** (recommended) and **conda** as package manager
- Automatic system checks (NVIDIA driver, GLIBC version)
- Support for multiple versions side by side
- Interactive step-by-step installation with verification

## What It Does

1. Select IsaacLab version
2. Select package manager (uv / conda)
3. Check system dependencies, NVIDIA driver, GLIBC
4. Setup package manager and create virtual environment
5. Install IsaacSim and PyTorch
6. Clone IsaacLab repo and install dependencies
7. Verify installation

## Known Issues

- When installing v1.4.1, if you encounter `No matching distribution found for rsl-rl`, edit `source/extensions/omni.isaac.lab_tasks/setup.py` line 46: change `"rsl-rl": ["rsl-rl@` to `"rsl-rl": ["rsl-rl-lib@`, then re-run the script.

**Hope this helps! Feel free to open an issue if you run into any problems.**

**Comment by RandomOakForest** [comment]
Thanks for posting this. If you could explain the one-liner for the community would be great.
